assigment4/assignment4.py

```diff
@@ -0,0 +1,54 @@
+class Tile:
+ 
+    def __init__(self, row, col, g):
+        self.ROW = row
+        self.COL = col
+        self.graph = g
+ 
+    def isSafe(self, i, j, visited):
+        return (i >= 0 and i < self.ROW and
+                j >= 0 and j < self.COL and
+                not visited[i][j] and self.graph[i][j])
+             
+    def DFS(self, i, j, visited):
+        #Direction arrays
+        rowNbr = [-1,0,-1,0];
```

Comment:
It might be a bit nicer with row/col offset pairs:

```
directions = [(-1, 0), (1, 0), (0, -1), (0, 1)]

...

for (di, dj) in directions:
  if self.isSafe(i + di, j + dj, visited):
    self.DFS(i + di, j + dj, visited)
```